spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -101,6 +101,10 @@ public FirestoreTemplate(FirestoreStub firestore, String parent, FirestoreClassM
 		this.mappingContext = mappingContext;
 	}
 
+	public <T> FirestoreReactiveOperations operationsWithParent(T parent) {
+		return new FirestoreTemplate(this.firestore, buildResourceName(parent), this.classMapper, this.mappingContext);
```

Comment:
I think you may need to invent a new abstraction here which implements `FirestoreReactiveOperations` and just has a reference to the subcollection and the `FirestoreTemplate` and then delegates calls to the firestoretemplate.

I think copying the whole `FirestoreTemplate` object is too heavyweight for just setting the root; feels like there is too much redundant information being copied each time the method is called. Another reason why it doesn't feel quite right is because you are also going to need to call all the `setter` methods to copy over the settings that were set outside the constructor to do this right.

Reply:
It only "copies" references. What do you `setter` do you mean?

Reply:
`FirestoreTemplate` has settings like buffer or useTokens which are not set through constructor but setters, these would have to be copied as well

Reply:
I guess for now you can just call the setters so that you set the properties of the child template in the same way as the parent; and if the future if we need to optimize we can do it later.